core/trino-main/src/test/java/io/trino/sql/query/TestColumnMask.java

```diff
@@ -209,7 +209,8 @@ public void testMultipleMasksOnSameColumn()
                 USER,
                 new ViewExpression(USER, Optional.empty(), Optional.empty(), "custkey * 2"));
 
-        assertThat(assertions.query("SELECT custkey FROM orders WHERE orderkey = 1")).matches("VALUES BIGINT '-740'");
+        // When there are multiple masks on the same column, the latter one overrides the previous ones
+        assertThat(assertions.query("SELECT custkey FROM orders WHERE orderkey = 1")).matches("VALUES BIGINT '740'");
```

Comment:
This is a change in semantics. Looking at this, I realize we never settled the question of why we'd ever have multiple masks on a single column: https://github.com/trinodb/trino/pull/11654#issuecomment-1079156689. That still doesn't make a lot of sense to me. cc @kokosing @ksobolew

Reply:
We can have multiple system access controls and each of them can return mask for a given column.

Reply:
How are the masks expected to be applied in such a case? There's no guarantee about the order in which the system access controls are invoked.

Also, you're talking about multiple system access controls, but the SystemAccessControl interface allows a *single* access control interface to return multiple masks for a given column. That does not make much sense -- if only one mask is to be applied, the interface should return a single mask.

Reply:
This question appeared while we were working on BIAC.

There are multiple situations in which more than one mask may apply to a single column. The simplest example is when there are two roles enabled and each of them applies a different column mask on a particular column. This immediately raises the question of ordering, of course. While it looks like Trino does not make such guarantees *explicitly*, the ordering is in practice deterministic as long as the access controls return them in a deterministic order. The masks and filters are returned and processed as `List`s, and this implies ordering.

In BIAC I made sure that the ordering is deterministic, even if not strictly defined, and we have tests that verify the application order. See #5611, which added ordering for grants, followed up by #5421, which separated masks and filters from grants while inheriting the ordering guarantee, and finally #5751, which fixed some gaps related to role grants. (Note that we also did this for row filters, which may look unnecessary at first glance; the reason is that some expressions may have side effect - like the canonical example of division by zero - which may or may not be exhibited differently thanks to short-circuiting.)

Reply:
Sorry, the comment I posted earlier, was not meant for this repository :(

Reply:
There are multiple situations in which more than one mask may apply to a single column. The simplest example is when there are two roles enabled and each of them applies a different column mask on a particular column. This immediately raises the question of ordering, of course. While it looks like Trino does not make such guarantees *explicitly*, the ordering is in practice deterministic as long as the access controls return them in a deterministic order. The masks and filters are returned and processed as `List`s, and this implies ordering. The engine has no trouble dealing with this and applies the masks iteratively.

Reply:
That still doesn't make a lot of sense to me. Masks are defined in terms of the values in the column. For example, if column is a string that contains only numbers, one might implement a mask function by parsing the number, doing some math (a hash, etc) on it and converting it back to a string. If the the engine applies the function on anything other than the original values (e.g., because another mask got injected in between), the mask will fail. This makes it hard for someone writing a mask function to reason about, as it requires non-local understanding of how every role, user and mask relate to each other.

If multiple masks are possible, it should be up to the connector to decide which one is the most appropriate and go with it. In the case of multiple system access controls returning masks, there are two options:
* Pick the first one. This can produce non-deterministic results depending on which order the access controls get called. It could change across restarts of the server
* Fail with an error indicating there's ambiguity.